Stratis.Bitcoin/BlockPulling/LookaheadBlockPuller.cs

```diff
@@ -403,14 +407,21 @@ private Block NextBlockCore(CancellationToken cancellationToken)
                     }
                     else
                     {
+                        // TODO: Race condition here - as soon as IsDownloading returns false here, 
+                        // another thread can change that (consider BlockPullerBehavior.Node_MessageReceived, 
+                        // just before the newly downloaded block is pushed, the above check for DownloadedBlocks 
+                        // will fail - block not pushed yet - but then IsDownloading here returns false), which 
+                        // will cause this thread to call AskBlocks for a block that was already downloaded. 
+                        // https://github.com/stratisproject/StratisBitcoinFullNode/issues/244
```

Comment:
IsDownloading is only called form one thread however AskBlocks I am not sure, but if we ask a block twice the map dictionary will throw.